pkg/logs/input/file/tailer.go

```diff
@@ -213,7 +224,7 @@ func (t *Tailer) forwardMessages() {
 		origin := message.NewOrigin(t.source)
 		origin.Identifier = identifier
 		origin.Offset = strconv.FormatInt(offset, 10)
-		origin.SetTags(t.tags)
+		origin.SetTags(append(t.tags, t.tagProvider.GetTags()...))
```

Comment:
Nits: Would be nice to cache `append(t.tags, t.tagProvider.GetTags()...)` and reallocate only on new tags